src/main/java/org/opensearch/performanceanalyzer/collectors/telemetry/RTFHeapMetricsCollector.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.performanceanalyzer.collectors.telemetry;
+
+import java.lang.management.MemoryUsage;
+import java.util.Map;
+import java.util.function.Supplier;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.performanceanalyzer.OpenSearchResources;
+import org.opensearch.performanceanalyzer.commons.collectors.PerformanceAnalyzerMetricsCollector;
+import org.opensearch.performanceanalyzer.commons.jvm.GCMetrics;
+import org.opensearch.performanceanalyzer.commons.jvm.HeapMetrics;
+import org.opensearch.performanceanalyzer.commons.metrics.AllMetrics;
+import org.opensearch.performanceanalyzer.commons.metrics.MetricsConfiguration;
+import org.opensearch.performanceanalyzer.commons.stats.metrics.StatExceptionCode;
+import org.opensearch.performanceanalyzer.commons.stats.metrics.StatMetrics;
+import org.opensearch.telemetry.metrics.Histogram;
+import org.opensearch.telemetry.metrics.MetricsRegistry;
+import org.opensearch.telemetry.metrics.tags.Tags;
+
+public class RTFHeapMetricsCollector extends PerformanceAnalyzerMetricsCollector {
+    private static final Logger LOG = LogManager.getLogger(RTFHeapMetricsCollector.class);
+    public static final int SAMPLING_TIME_INTERVAL =
+            MetricsConfiguration.CONFIG_MAP.get(RTFHeapMetricsCollector.class).samplingInterval;
+    private Histogram gcCollectionEventMetrics;
+    private Histogram gcCollectionTimeMetrics;
+    private Histogram heapUsedMetrics;
+    private MetricsRegistry metricsRegistry;
+    private final String memTypeAttributeKey = "mem_type";
+    private boolean metricsInitialised;
+
+    public RTFHeapMetricsCollector() {
+        super(
+                SAMPLING_TIME_INTERVAL,
+                "RTFHeapMetricsCollector",
+                StatMetrics.HEAP_METRICS_COLLECTOR_EXECUTION_TIME,
+                StatExceptionCode.HEAP_METRICS_COLLECTOR_ERROR);
+        this.metricsInitialised = false;
+    }
+
+    @Override
+    public void collectMetrics(long startTime) {
+        metricsRegistry = OpenSearchResources.INSTANCE.getMetricsRegistry();
+        if (metricsRegistry == null) {
+            LOG.error("could not get the instance of MetricsRegistry class");
+            return;
+        }
+
+        initialiseMetricsIfNeeded();
+        GCMetrics.runGCMetrics();
+        LOG.debug("Executing collect metrics for RTFHeapMetricsCollector");
+        recordMetrics();
+    }
+
+    private void initialiseMetricsIfNeeded() {
+        if (metricsInitialised == false) {
```

Comment:
I am wondering should we have a flag or not null check? Because if we may miss the setting flag to null then it will be an issue and hard to track.

Reply:
Would like to discuss this one.

Reply:
Discussed offline. We should be good with this implementation for now. We will revisit if needed.